### PR TITLE
[WIP] fixing scaladoc warnings reported by jenkins

### PR DESCRIPTION
**Pull request by wlk**
Could you start jenkins build to see if I didn't miss anything so far?

I'm unable to get the same warnings when running locally

### CODE REVIEW

**Comment by ktoso** [comment]
These are the options jenkins runs sbt with:

`-Dakka.genjavadoc.enabled=true -Dsbt.ivy.home=/localhome/jenkinsakka/.ivy2 -Dakka.build.M2Dir=/localhome/jenkinsakka/.m2/repository -Dakka.test.timefactor=1 -Dakka.cluster.assert=on -Dsbt.override.build.repos=false -Dakka.test.multi-node=true -Dakka.test.multi-node.targetDirName=/localhome/jenkinsakka/$JOB_NAME/$BUILD_NUMBER -Dakka.test.multi-node.java=/usr/lib/jvm/java-1.8.0-openjdk/bin/java -Dmultinode.Xms64M -Dmultinode.Xmx128M -Dmultinode.XX:MetaspaceSize=128M -Dmultinode.XX:+PrintGCDetails -Dmultinode.XX:+PrintGCTimeStamps
`

**Comment by wlk** [comment]
yeah, I know. It's weird that I'm seeing different output, I'm generating docs like that: `sbt doc -Dakka.genjavadoc.enabled=true -Dakka.scaladoc.diagrams=false` - setting those 2 things because they are related to generating scala/javadoc. Is there something else I'm missing?

This is doc output generated from current master: http://pastebin.com/raw/C1ZZbG6S 
If you grep for "reference not found" you won't find anything, but on jenkins, for example here: https://jenkins.akka.io:8498/view/Dashboard/job/pr-validator-per-commit-jenkins/5455/consoleFull you will find 220 warnings

**Comment by ktoso** [comment]
Are you using JDK8 (jenkins is)

**Comment by wlk** [comment]
```
java version "1.8.0_74"
Java(TM) SE Runtime Environment (build 1.8.0_74-b02)
Java HotSpot(TM) 64-Bit Server VM (build 25.74-b02, mixed mode)
```
sbt 0.13.11

scala 2.11.8

**Comment by patriknw** [comment]
OK TO TEST

**Comment by wlk** [comment]
There are still warnings after jenkins build, I'll review them and prepare new commit (probably in few days)

File: akka-http/src/main/scala/akka/http/javadsl/server/Marshaller.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@
 package akka.http.javadsl.server
 
 /**
- * A marker trait for a marshaller that converts a value of type [[T]] to an
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/javadsl/server/Marshaller.scala]: hehe silly one :)

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**